### DISCUSSION TITLE
What's the aim of function betadisper?

### DISCUSSION THREAD

**Discussion by Johnme24** [vegandevs/vegan#476] [Q&A]
I am very new to vegan package. I wonder what the funcion betadisper aims for? Since the help(betadisper) pointed out that the betadisper is a multivariate analogue of Levene's test for homogeneity of variances. Do I need to perform this function before the PERMANOVA with the function adonis().It looks like this was not seen in other cases as far as I know. I really feel confused. 
In addition, I wanna to  know how to dopairwise comparison for PERMANOVA, if the factor was two kinds of grouping variables. How to fix the the code as I copied from the net. factors=list(var1,var2) does not work. 
Much thanks

```r
 pairwise.adonis <- function(x,factors, 
                    sim.method = 'bray', p.adjust.m ='bonferroni'){
  library(vegan)
  co = combn(unique(factors),2)
  pairs = c()
  F.Model =c()
  R2 = c()
  p.value = c()
  for(elem in 1:ncol(co)){
    ad = adonis(x[factors %in% c(co[1,elem],co[2,elem]),] ~ factors[factors %in% c(co[1,elem],co[2,elem])] , method =sim.method);
    pairs = c(pairs,paste(co[1,elem],'vs',co[2,elem]));
    F.Model =c(F.Model,ad$aov.tab[1,4]);
    R2 = c(R2,ad$aov.tab[1,5]);
    p.value = c(p.value,ad$aov.tab[1,6])
  }
  p.adjusted = p.adjust(p.value,method=p.adjust.m)
  pairw.res = data.frame(pairs,F.Model,R2,p.value,p.adjusted)
  return(pairw.res)
}
```

**Comment by gavinsimpson** [comment]
The rationale is given in *inter alia* [Anderson & Walsh (2013)](http://doi.org/10.1890/12-2010.1)

For the other part of your question, perhaps ask on StackOverflow or the person who wrote the code you are trying to modify?

**Comment by FVFaleiro** [comment]
Here you find the pairwise adonis: https://github.com/pmartinezarbizu/pairwiseAdonis and here a discussion about it: https://www.researchgate.net/post/How_can_I_do_PerMANOVA_pairwise_contrasts_in_R.
